mono/metadata/w32file-unix.c

```diff
@@ -4402,7 +4402,7 @@ mono_w32file_get_disk_free_space (const gunichar2 *path_name, guint64 *free_byte
 typedef struct {
 	guint32 drive_type;
 #if __linux__
-	const long fstypeid;
+	long fstypeid;
```

Comment:
From http://man7.org/linux/man-pages/man2/statfs.2.html#NOTES
> The __fsword_t type used for various fields in the statfs structure
>       definition is a glibc internal type, not intended for public use.
>       This leaves the programmer in a bit of a conundrum when trying to
>       copy or compare these fields to local variables in a program.  Using
>       unsigned int for such variables suffices on most systems.

So maybe we should just use `unsigned long` here?

Reply:
So..this is gnarly stuff actually. We have to have some value we store, and then later compare it to some value we get back. We need to ensure that sign extension works the same.

I think the best options are:
1. global: _fsword_t 

2. global: almost anything, int, int32, uint, uint32, long (casted), ulong, etc.
But don't use the value directly.
Have a local struct statfs, store the global into statfs.f_type, and then compare to the other statfs.f_type. Having the full struct statfs and storing into a temporary is a C form of casting to `decltype`.

3. Or what I showed, just cast, but put an #if around so it is for 32bit long, in order to not change semantics.